src/storage/src/source/reclock.rs

```diff
@@ -587,11 +589,13 @@ impl ReclockOperator {
         if (now % self.update_interval_ms) != 0 {
             new_ts += self.update_interval_ms;
         }
+        let new_ts: Timestamp = new_ts.try_into().expect("must fit");
         let upper_ts = self.upper.as_option().expect("no more timestamps to mint");
         if upper_ts <= &new_ts {
             Ok(new_ts)
         } else {
-            Err(Duration::from_millis(upper_ts - now))
+            let upper: u64 = upper_ts.into();
+            Err(Duration::from_millis(upper - now))
```

Comment:
This is also maybe for a future PR, but `now` made me think of it. We also have the `EpochMillis` type which is just an alias for `u64`. I wonder how many uses of that type should really be using the opaque `Timestamp` type.

Reply:
Agree. We have a variety of things that interact together: mz_repr::Timestamps, Durations, and unix epoch milliseconds. Getting clear type boundaries between these is probably useful.